src/components/enterprise-redirects/EnterpriseLearnerFirstVisitRedirect.jsx

```diff
@@ -2,23 +2,24 @@ import React, { useEffect } from 'react';
 import { Redirect, useParams } from 'react-router-dom';
 import Cookies from 'universal-cookie';
 
+export const isFirstDashboardPageVisit = () => {
+  const cookies = new Cookies();
+
+  const hasUserVisitedDashboard = cookies.get('has-user-visited-learner-dashboard');
+  return !hasUserVisitedDashboard;
+};
+
 const EnterpriseLearnerFirstVisitRedirect = () => {
   const { enterpriseSlug } = useParams();
-
   const cookies = new Cookies();
 
-  const isFirstVisit = () => {
-    const hasUserVisitedDashboard = cookies.get('has-user-visited-learner-dashboard');
-    return !hasUserVisitedDashboard;
-  };
-
   useEffect(() => {
-    if (isFirstVisit()) {
+    if (isFirstDashboardPageVisit()) {
       cookies.set('has-user-visited-learner-dashboard', true, { path: '/' });
     }
   });
```

Comment:
This `useEffect` doesn't have a dependency array, which means it will execute on every component re-render.